src/components/NewsGrid/index.jsx

```diff
@@ -35,20 +39,28 @@ const NewsGrid = () => {
                             columns={{base: 1, sm: 2, md: 3, lg: 4}}
                             spacing={3}>
                             {
-                                allNewsSources.map((newsSource, index) => (
-                                    <Draggable key={newsSource} draggableId={newsSource} index={index}>
+                                visibleNewsCards.map((source, index) => (
+                                    <Draggable key={source.id} draggableId={source.id} index={index}>
                                         {(draggableProvided) => (
                                             <div
                                                 {...draggableProvided.draggableProps}
-                                                ref={draggableProvided.innerRef}
-                                                {...draggableProvided.dragHandleProps}>
+                                                ref={draggableProvided.innerRef}>
                                                 <NewsCard
-                                                    key={newsSource}
-                                                    source={newsSource}/>
+                                                    key={source.id}
+                                                    source={source}>
+                                                    <Flex>
```

Comment:
Please try to see if this component can be split into multiple components

Reply:
Looks fine to me

Reply:
as discussed in meeting, leaving as is due to complications with react-beautiful-dnd